lib/src/Collection.cpp

```diff
@@ -351,3 +472,764 @@ shared_ptr<DocumentIterator> Collection::QueryDocuments(
 {
 	return this->QueryDocumentsAsync(query, page_size).get();
 }
+
+Concurrency::task<shared_ptr<Trigger>> Collection::CreateTriggerAsync(
+	const wstring& id,
+	const wstring& body,
+	const TriggerOperation& triggerOperation,
+	const TriggerType& triggerType) const
+{
+	http_request request = CreateRequest(
+		methods::POST,
+		RESOURCE_PATH_TRIGGERS,
+		this->resource_id(),
+		this->document_db_configuration()->master_key());
+	request.set_request_uri(this->self() + triggers_);
+	
+	value body_;
+	body_[DOCUMENT_ID] = value::string(id);
+	body_[BODY] = value::string(body);
+	switch (triggerOperation) {
+	case TriggerOperation::ALL:
+		body_[TRIGGER_OPERATION] = value::string(L"All");
+		break;
+	case TriggerOperation::CREATE:
+		body_[TRIGGER_OPERATION] = value::string(L"Create");
+		break;
+	case TriggerOperation::UPDATE:
+		body_[TRIGGER_OPERATION] = value::string(L"Update");
+		break;
+	case TriggerOperation::REPLACE:
+		body_[TRIGGER_OPERATION] = value::string(L"Replace");
+		break;
+	case TriggerOperation::DEL:
+		body_[TRIGGER_OPERATION] = value::string(L"Delete");
+		break;
+	}
+	switch (triggerType) {
+	case TriggerType::PRE:
+		body_[TRIGGER_TYPE] = value::string(L"Pre");

```

Comment:
you have those mappings between enum and string all over the place...ideally, they should be at one place and one place only